utoipa-swagger-ui/build.rs

```diff
@@ -32,10 +33,18 @@ fn main() {
     let zip_path = [&target_dir, &zip_filename].iter().collect::<PathBuf>();
 
     if !zip_path.exists() {
-        println!("start download to : {:?}", zip_path);
-        download_file(&url, zip_path.clone()).unwrap();
+        if url.starts_with("http://") || url.starts_with("https://") {
+            println!("start download to : {:?}", zip_path);
+            download_file(&url, zip_path.clone()).unwrap();
+        } else if url.starts_with("file://") {
+            let file_path = url.replace("file://", "");
+            println!("start copy to : {:?}", zip_path);
+            fs::copy(file_path, zip_path.clone()).unwrap();
```

Comment:
No need to copy the file here, right? Wouldn't it be enough to set `zip_path = file_path;`?

Reply:
In principle yes, but I prefer having minimal changes between the 2 cases. Here once the download/copy operation is performed, the next steps are exactly the same: Everthing happens in `target_dir`.

Reply:
Fwiw, this also means that if the zip file changes then that won't take effect before you `cargo clean` (or remove the cached file manually).

Reply:
Hm. I would then guess it would be useful to make the file type of `SWAGGER_UI_DOWNLOAD_URL` to just rely on external Swagger UI. Then it is not tied to the _cargo lifecycle_ and can be externally updated.

Reply:
Note that if the file name changes (via env var update), then no need to `cargo clean`. And these `.zip` files are meant to be immutable, straight from official source.

Reply:
Oh, yeah indeed because of this: `println!("cargo:rerun-if-changed={:?}", zip_path.clone());`

Reply:
and just `if !zip_path.exists() {...}` tests if download/copy is necessary.

Reply:
The `rerun-if-changed` detects if the _cached zip_ changes, not the source zip fwiw.

Reply:
So to summarize:
+ one can change a `swagger-xxx.zip` file via env var `SWAGGER_UI_DOWNLOAD_URL` with `file://` 
+ these file, typically from [official releases](https://github.com/swagger-api/swagger-ui/tags), are uniquely identified by their name.
+ if for some reason somebody creates their own, they just have to change filename, change `SWAGGER_UI_DOWNLOAD_URL` correspondingly and `cargo build` - if they use a name the same name it will not be downloaded/copied again.
+ to modifiy the unzipped contents one can use env var `SWAGGER_UI_OVERWRITE_FOLDER` for selective overwrites - this will be applied in all cases.

This a fine grained config allows most if not all custom needs, I think.